jsFiles/design.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * This function changes the color of the background-sky,
+ * according to the actual time, sunset and sunrise time.
+ * Additionally it changes the letter of the card and the filter of the house
+ * according to the color of the sky.
+ * @param {number} sunrise - time stamp with time info
+ * @param {number} sunset - time stamp with time info
+ * @param {number} offsetTime - offset time refering to the time zone
+ */
+function skyColor(sunrise, sunset, offsetTime) {
+  const timeInMinutes = timeConversion(
+    currentDateUserLocation,
+    offsetTime,
+    true
+  );
+  const sunriseInMinutes = timeConversion(sunrise, offsetTime, true);
+  const sunsetInMinutes = timeConversion(sunset, offsetTime, true);
+
+  const startSunrise = sunriseInMinutes - 60;
+  const endSunrise = sunriseInMinutes + 60;
+  const aroundSunrise = endSunrise - startSunrise;
+
+  const startSunset = sunsetInMinutes - 60;
+  const day = startSunset - endSunrise;
+  const endSunset = sunsetInMinutes + 60;
+
+  const aroundSunset = endSunset - startSunset;
+  // A day has 1440 min: 60*24
+
+  // night
+  if (
+    timeInMinutes < startSunrise ||
+    (timeInMinutes > endSunset && timeInMinutes <= 1440)
+  ) {
+    cardBody.style.color = "#fff";
+    imageHouse.style.filter = "invert(0.9)";
+    cardBody.style.backgroundImage = skyColors[1];
+    console.log("color 1 is on ");
+  }
+  // end night
+  // sunrise
+  else if (timeInMinutes < startSunrise + aroundSunrise / 3) {
+    cardBody.style.backgroundImage = skyColors[2];
+    cardBody.style.color = "#fff";
+    imageHouse.style.filter = "sepia(0.6)";
+    console.log("color 2 is on ");
+  }
+  // section with color: black;  & filer: none
+  else if (timeInMinutes < startSunset + (aroundSunset / 3) * 2) {
```

Comment:
just curious what these magic numbers mean? 3 and 2

Reply:
:) so I divided the sunrise and sunset in 3 parts, each part has its own color ... the second of those parts *2.  I use to have many colors so it was /8 *2, and when I simplified the function to use less colors I stood by this way of writing.